frontend/src/metabase/query_builder/components/NativeQueryEditor/NativeQueryEditor.tsx

```diff
@@ -723,6 +724,17 @@ export class NativeQueryEditor extends Component<
     );
   };
 
+  formatQuery = async () => {
+    const engine = this.props.question?.database?.()?.engine;
+    const queryText = this._editor?.getValue();
```

Comment:
```suggestion
    const query = question.query();
    const engine = Lib.engine(query);
    const queryText = Lib.rawNativeQuery(query);
```

We ~~can~~ should? use MLv2 going forward. Just make sure you guard against `question` potentially being undefined, and all this should work just fine :)